lib/screens/CourseDirectoryScreen.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:we_find/model/modelTranslate.dart';
+
+import 'StudyModuleScreen.dart';
+
+// TODO: This is just a copy of StudyModuleScreen with some changes
+// if you know how to fix this feel free.
```

Comment:
The course directory is basically the root of a module tree, therefore the ideal solution would be to create a single widget that represents a module and handle the recursive rendering of its submodules. The base case of the recursion would be when a module has no further modules. In this case, the courses/exams of that module could be rendered.

The code that you wrote gives a great foundation to this! I think it would be a viable solution to refactor `StudyModuleScreen` to use a single widget, `StudyModuleView` and reuse it in `StudyModuleScreen` and `CourseDirectoryScreen` as well.

Reply:
I will look into this before merging and try to push a solution ASAP.

Reply:
Done.